### PR TITLE
CLI Experimental api removal

### PR DESCRIPTION
**Pull request by drcariel**
https://github.com/apache/incubator-openwhisk/issues/2288

- CLI Go code
  - [x] Remove experimental command, associated help, associated wski18n strings
  - [x] Replace V2 labeled structures and functions. 
- Tests
  - ApiGwRoutemgmtActionTests
    - [x] Remove experimental functions.
    - [x] Re-label V2 items (remove V2) - methods, test labels, etc.
    - [x] Delete entire API Gateway routemgmt action parameter validation test suite
  - ApiGwTests
    - [x] Remove all experimental tests and helper functions
  - ApiGwEndToEndTests
    - [x] Remove api-experimental test suite
- Deployment
    - [x] installRouteMgmt.sh task. Remove installation of routemgmt package and associated actions

- LAST
  - [x] WskPropsV2

### CODE REVIEW

**Comment by rabbah** [comment]
@drcariel i highly recommend you pick up my pr https://github.com/apache/incubator-openwhisk/pull/2493 esp for the test related changes.

**Comment by drcariel** [comment]
@rabbah we can discuss in https://github.com/drcariel/incubator-openwhisk/pull/2 if you'd like.

@mdeuser

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala [hunk 1]
```diff
@@ -1035,7 +607,7 @@ class ApiGwTests
             rr.stderr should include("does not exist")
         }
         finally {
-            val deleteresult = apiDeleteExperimental(basepathOrApiName = testbasepath, expectedExitCode = DONTCARE_EXIT)
+            val deleteresult = apiDelete(basepathOrApiName = testbasepath, expectedExitCode = DONTCARE_EXIT)
```
Comment (mdeuser) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: good catch!

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala [hunk 2]
```diff
@@ -1275,4 +847,4 @@ class ApiGwTests
             var deleteresult = apiDelete(basepathOrApiName = testbasepath, expectedExitCode = DONTCARE_EXIT)
         }
     }
-}
+}
```
Comment (mdeuser) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: newline

File: tests/src/test/scala/whisk/core/cli/test/WskBasicUsageTests.scala [hunk 3]
```diff
@@ -834,41 +834,41 @@ class WskBasicUsageTests
 
     behavior of "Wsk api"
 
-    it should "reject an api commands with an invalid path parameter" in {
```
Comment (mdeuser) [inline tests/src/test/scala/whisk/core/cli/test/WskBasicUsageTests.scala]: remove.  the "api" test is already in ApiGwTests.scala

File: tools/cli/go-whisk-cli/commands/rule.go [hunk 4]
```diff
@@ -388,6 +388,19 @@ var ruleListCmd = &cobra.Command{
                     map[string]interface{}{"name": getClientNamespace(), "err": err})
             werr := whisk.MakeWskErrorFromWskError(errors.New(errStr), err, whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
             return werr
+        } else {
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/rule.go]: not related to api-experimental removal..?

File: tools/cli/go-whisk-cli/commands/util.go [hunk 5]
```diff
@@ -332,7 +332,7 @@ func printRuleList(rules []whisk.Rule) {
     fmt.Fprintf(color.Output, "%s\n", boldString("rules"))
     for _, rule := range rules {
         publishState := wski18n.T("private")
-        fmt.Printf("%-70s %s\n", fmt.Sprintf("/%s/%s", rule.Namespace, rule.Name), publishState)
+        fmt.Printf("%-70s %-20s %s\n", fmt.Sprintf("/%s/%s", rule.Namespace, rule.Name), publishState, rule.Status)
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/util.go]: not related to api-experimental removal..?

File: tools/cli/go-whisk-cli/commands/rule.go (hunk 4 above)
Comment (drcariel) [inline tools/cli/go-whisk-cli/commands/rule.go]: This is the recent commit to apache master, I probably didn't do this right. I can back it out if you'd like

File: tools/cli/go-whisk-cli/commands/util.go (hunk 5 above)
Comment (drcariel) [inline tools/cli/go-whisk-cli/commands/util.go]: same commit to master as mentioned above

File: tests/src/test/scala/whisk/core/apigw/actions/test/ApiGwRoutemgmtActionTests.scala [hunk 6]
```diff
@@ -342,7 +212,7 @@ class ApiGwRoutemgmtActionTests
         return matches
     }
 
-    behavior of "API Gateway routemgmt action parameter validation"
```
Comment (drcariel) [inline tests/src/test/scala/whisk/core/apigw/actions/test/ApiGwRoutemgmtActionTests.scala]: I am pretty sure that is how I implemented this change. Github is picking up the deletion and removal of V2 tags as a modify. Threw me off at first as well

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala (hunk 1 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: yea, I assume this was a bug right?

File: tests/src/test/scala/system/basic/WskRuleTests.scala [hunk 7]
```diff
@@ -371,4 +370,23 @@ class WskRuleTests
             }
     }
 
+    it should "disable a rule and check its status is displayed when listed" in withAssetCleaner(wskprops) {
```
Comment (drcariel) [inline tests/src/test/scala/system/basic/WskRuleTests.scala]: recent commit to master again

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala (hunk 2 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: done

File: tests/src/test/scala/apigw/healthtests/ApiGwEndToEndTests.scala [hunk 8]
```diff
@@ -244,4 +174,4 @@ class ApiGwEndToEndTests
             )
         }
     }
-}
+}
```
Comment (drcariel) [inline tests/src/test/scala/apigw/healthtests/ApiGwEndToEndTests.scala]: done

File: ansible/roles/routemgmt/files/installRouteMgmt.sh [hunk 9]
```diff
@@ -48,30 +48,6 @@ zip -j "$OPENWHISK_HOME/core/routemgmt/getApi/getApi.zip" "$OPENWHISK_HOME/core/
...
--a description 'Create an API' \
---kind nodejs:default \
--a web-export true -a final true -a require-whisk-auth true
-
-$WSK_CLI -i --apihost "$APIHOST" action update --auth "$AUTH" "$NAMESPACE/routemgmt/deleteApi" "$OPENWHISK_HOME/core/routemgmt/deleteApi/deleteApi.zip" \
--a description 'Delete the API' \
---kind nodejs:default \
--a web-export true -a final true -a require-whisk-auth true
-
-echo Installing apimgmt package
```
Comment (drcariel) [inline ansible/roles/routemgmt/files/installRouteMgmt.sh]: a bit confused here, do you want me to move the Installation apimgmt package below the .zip files?

Comment (mdeuser) [inline ansible/roles/routemgmt/files/installRouteMgmt.sh]: remove the `routemgmt` package and its associated actions.  keep the `apimgmt` package.

Comment (drcariel) [inline ansible/roles/routemgmt/files/installRouteMgmt.sh]: yes, that is exactly what I have done already

**Comment by drcariel** [comment]
most recent passing build: PG2~1884

File: ansible/roles/routemgmt/files/installRouteMgmt.sh (hunk 9 above)
Comment (mdeuser) [inline ansible/roles/routemgmt/files/installRouteMgmt.sh]: ok.  the git diff was probably messing me up. :-)

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala (hunk 1 above)
Comment (mdeuser) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: a couple more "V2" related items
- testswaggerdoc1V2, testswaggerdoc2V2 
- WskPropsV2.. save this for last, but i think this can replace the existing WskProps definition in the same file.

File: tests/src/test/scala/whisk/core/cli/test/WskBasicUsageTests.scala [hunk 10]
```diff
@@ -1085,13 +1004,6 @@ class WskBasicUsageTests
         val invalidParamFileMsg = "An argument must be provided for '-P'"
         val invalidAnnotFileMsg = "An argument must be provided for '-A'"
         val invalidArgs = Seq(
-            (Seq("action", "create", "actionName", "-p"), invalidParamMsg),
```
Comment (mdeuser) [inline tests/src/test/scala/whisk/core/cli/test/WskBasicUsageTests.scala]: block of changes seem unrelated to api-experimental removal

**Comment by drcariel** [comment]
another passing build: PG1~1880

File: tests/src/test/scala/whisk/core/cli/test/WskBasicUsageTests.scala (hunk 10 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskBasicUsageTests.scala]: must have copied something incorrectly over, adding back in

**Comment by drcariel** [comment]
another build: PG1~1881

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala (hunk 1 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: testswaggerdocs have been modified and renamed

**Comment by drcariel** [comment]
PG3~943

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala (hunk 1 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: wskprops has been updated as well

Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: whiskprops is now switched out as well, however, may need to figure out what's going on in WskConfigTests.scala. I didn't see any tests for V2, so I did my best to modify the existing ones (to get around compiler errors at least). I have a feeling the tests will fail

**Comment by drcariel** [comment]
Latest build PG2~1898

File: tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala (hunk 1 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/ApiGwTests.scala]: tests are passing

File: tests/src/test/scala/common/Wsk.scala [hunk 11]
```diff
@@ -71,16 +71,6 @@ import whisk.utils.retry
 
 case class WskProps(
     authKey: String = WhiskProperties.readAuthKey(WhiskProperties.getAuthFileForTesting),
-    cert: String = WhiskProperties.getFileRelativeToWhiskHome("ansible/roles/nginx/files/openwhisk-client-cert.pem").getAbsolutePath,
```
Comment (rabbah) [inline tests/src/test/scala/common/Wsk.scala]: Restore cert and key.

File: tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala [hunk 12]
```diff
@@ -149,9 +149,7 @@ class WskConfigTests
                 // Send request to https://<apihost>/api/v1/namespaces, wsk client passes client certificate to nginx, nginx will
                 // verify it by client ca's openwhisk-client-ca-cert.pem
                 val stdout = wsk.cli(Seq("property", "set", "-i", "--apihost", wskprops.apihost, "--auth", wskprops.authKey,
-                    "--cert", wskprops.cert, "--key", wskprops.key, "--namespace", namespace), env = env).stdout
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: Undo these.

Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: wskprops.cert and wskprops.key will not compile anymore, the original wskprops has been removed and updated to use V2 implementation.

File: tests/src/test/scala/common/Wsk.scala (hunk 11 above)
Comment (drcariel) [inline tests/src/test/scala/common/Wsk.scala]: cert was only available in WskProps, not WskPropsV2. Why would I add it back in?

File: tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala (hunk 12 above)
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: i dont understand what cert and key have to do with removing api experimental.
they are not part of the api gw work.

File: tests/src/test/scala/common/Wsk.scala (hunk 11 above)
Comment (rabbah) [inline tests/src/test/scala/common/Wsk.scala]: WskPropsV2 was added for api gw - remove it.
WskProps is the correct class to retain.

File: tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala (hunk 12 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: Not so much what cert and key have to do with experimental, but what wskProps and WskPropsV2 have to do with the removal. The original wskprops was used by experimental (exclusively?), WskPropsV2 was used by non-experimental. @mdeuser asked me to clean up the 2 so now V2 is the only WskProps. Mark will have a better idea of exactly what this touches

Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: WskProps is the class to retain - you should merge v2 into it. Not the other way around.
Please restore cert and key.

Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: @ningyougang FYI above change - when these were added to WskProps, I suspected this would happen since they appear unused with respect to the tests. There is another way to preserve your test without modifying WskProps - perhaps we should do that.

Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: @drcariel not that you in fact changed the test to get it to pass. This is regressing functionality.

Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: I changed to test to compile, but yes, I will make the change

Comment (mdeuser) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: jsut to clarify.  WskPropsV2 was used exclusively by API GW tests (needed the additional api gw token property plus a method i think).  it's really not api-experimental specific.  i thought this was a good time to consolidate back into a single WskProps..

File: tests/src/test/scala/common/Wsk.scala (hunk 11 above)
Comment (drcariel) [inline tests/src/test/scala/common/Wsk.scala]: done, only question I have is about the apihost, they don't seem too different, but which one should I keep?

File: tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala (hunk 12 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskConfigTests.scala]: should be fixed now

File: tests/src/test/scala/common/Wsk.scala (hunk 11 above)
Comment (rabbah) [inline tests/src/test/scala/common/Wsk.scala]: they're the same value - i dont understand the question can you clarify?

Comment (drcariel) [inline tests/src/test/scala/common/Wsk.scala]: ah yes, I was confusing myself with some of the syntax. Apologies

**Comment by drcariel** [comment]
passing build: PG2~1904

**Comment by mdeuser** [comment]
@drcariel - there were 6 api-experimental tests removed from BasicUsageTests.  most of these are applicable to the api command.  please port these into the ApiGwTests.scala module, but only if the test is not already in that module.  for example, i don't think there's an existing test for "invalid verb" checking.

**Comment by drcariel** [comment]
PG2~1907

**Comment by drcariel** [comment]
PG3~968

**Comment by drcariel** [comment]
PG4~635

**Comment by drcariel** [comment]
PG1~1906 All up to date

**Comment by mdeuser** [comment]
@rabbah - is pr OK now?

**Comment by drcariel** [comment]
PG1~1918
